src/data/support.lean

```diff
@@ -0,0 +1,124 @@
+/-
+Copyright (c) 2020 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import data.set.basic algebra.field order.conditionally_complete_lattice
+  algebra.big_operators
+
+/-!
+# Support of a function
+
+In this file we define `function.support f = {x | f x ≠ 0}` and prove its basic properties.
+-/
+
+universes u v w x y
+
+open set
+namespace function
+
+variables {α : Type u} {β : Type v} {ι : Sort w} {A : Type x} {B : Type y}
+
+/-- `support` of a function is the set of points `x` such that `f x ≠ 0`. -/
+def support [has_zero A] (f : α → A) : set α := {x | f x ≠ 0}
+
+lemma nmem_support [has_zero A] {f : α → A} {x : α} :
+  x ∉ support f ↔ f x = 0 :=
+classical.not_not
+
+lemma support_binop_subset [has_zero A] (op : A → A → A) (op0 : op 0 0 = 0) (f g : α → A) :
+  support (λ x, op (f x) (g x)) ⊆ support f ∪ support g :=
+λ x hx, classical.by_cases
+  (λ hf : f x = 0, or.inr $ λ hg, hx $ by simp only [hf, hg, op0])
+  or.inl
+
+lemma support_add [add_monoid A] (f g : α → A) :
+  support (λ x, f x + g x) ⊆ support f ∪ support g :=
+support_binop_subset (+) (zero_add _) f g
+
+@[simp] lemma support_neg [add_group A] (f : α → A) :
+  support (λ x, -f x) = support f :=
+set.ext $ λ x, not_congr neg_eq_zero
+
+lemma support_sub [add_group A] (f g : α → A) :
+  support (λ x, f x - g x) ⊆ support f ∪ support g :=
+support_binop_subset (has_sub.sub) (sub_self _) f g
+
+@[simp] lemma support_mul [domain A] (f g : α → A) :
```

Comment:
Could you please add `support_mul'` for `group_with_zero`? They also don't have zero divisors.

Reply:
I wonder why `no_zero_divisors` doesn't `extend mul_zero_class`. Are there any important examples of `no_zero_divisors` without `0 * x = 0`?